packages/net/net_address.pony

```diff
@@ -108,18 +101,90 @@ class val NetAddress is Equatable[NetAddress]
       recover String.from_cstring(consume serv) end)
 
   fun eq(that: NetAddress box): Bool =>
-    (this.length == that.length)
-      and (this.family == that.family)
-      and (this.port == that.port)
+      (this._family == that._family)
+      and (this._port == that._port)
       and (host_eq(that))
-      and (this.scope == that.scope)
+      and (this._scope == that._scope)
 
   fun host_eq(that: NetAddress box): Bool =>
     if ip4() then
-      this.addr == that.addr
+      this._addr == that._addr
     else
-      (this.addr1 == that.addr1)
-        and (this.addr2 == that.addr2)
-        and (this.addr3 == that.addr3)
-        and (this.addr4 == that.addr4)
+      (this._addr1 == that._addr1)
+        and (this._addr2 == that._addr2)
+        and (this._addr3 == that._addr3)
+        and (this._addr4 == that._addr4)
     end
+
+  fun length() : U8 =>
+    """
+    For platforms (OSX/FreeBSD) with `length` field as part of 
+    its `struct sockaddr` definition, returns the `length`. 
+    Else (Linux/Windows) returns the size of `sockaddr_in` or `sockaddr_in6`.
+    """
+
+    ifdef linux or windows then
+      (@ponyint_address_length[U32](this)).u8()
+    else
+      ifdef bigendian then
```

Comment:
Everything looks good, i am just not 100% sure about this check. Could we have a testcase that verifies that the methods actually return what they should?